### PR TITLE
Improve cromwell monitoring script

### PR DESCRIPTION
**Pull request by TedBrookings**
* Added new cromwell monitoring script:
  /scripts/cromwell/cromwell_monitoring_script2.sh
  Outputs resource log in tabular format that 
  * uses less disk space
  * (is thus hopefully more likely to de-localize after job completion)
  * is easier to visually scan
  * is easier to ingest and process via script
* Added new script to extract summary info from new logs:
  /scripts/cromwell/get_cromwell_resource_usage2.sh 
  This script can process local or remote logs, and can select logs via
  cloud path, local path, or cromwell workflow ID

### CODE REVIEW

File: scripts/cromwell/get_cromwell_resource_usage2.sh [hunk 1]
```diff
@@ -0,0 +1,339 @@
...
+        | tail -n1 \
+    )
+fi
+1>&2 echo "WORKFLOW_DIR=$WORKFLOW_DIR"
+
+function get_monitor_logs() {
+    TOP_DIR=$1
+    REMOTE=$2
+    if $REMOTE; then
+      gsutil -m ls "$TOP_DIR/**monitoring.log" 2>/dev/null || echo ""
```
[the remark this answers is not in the document]
Comment (TedBrookings) [reply scripts/cromwell/get_cromwell_resource_usage2.sh:107]: I added code to remove trailing slashes from the input WORKFLOW_INFO when it's a cloud or local path. I tested it both with and without extra slashes and it appeared to work.

The other error you mentioned was: if you supplied options that bypasses the (no input arguments) check at the beginning of the script. If you forgot to supply WORKFLOW_INFO, it default to empty, isn't a path, and gets interpreted as an empty WORKFLOW_ID. Cromshell gets the last workflow when you pass it an empty workflow ID. I addressed this by checking for an empty WORKFLOW_INFO and throwing an error. I also moved around some (trivial) output to stderr to make errors more clear.

**Comment by TedBrookings** [comment]
@epiercehoffman, following our offline conversation, I added an option to omit the second "units" header line, and I altered the monitoring script to make the header column names unique. I'm happy to wait until you have a chance to test these changes (I did but your test routine is doubtlessly different from mine). If you want me to just go ahead and merge, I can do that too.

File: scripts/cromwell/get_cromwell_resource_usage2.sh [hunk 2]
```diff
@@ -50,6 +53,8 @@ for ((i=1; i<=$#; ++i)); do
         exit 0
     elif [[ ${!i} =~ ^-+(r|raw-output) ]]; then
         RAW_OUTPUT=true
+    elif [[ ${!i} =~ ^-+(u|hide-units) ]]; then
```
Comment (epiercehoffman) [inline scripts/cromwell/get_cromwell_resource_usage2.sh:56]:
```suggestion
    elif [[ ${!i} =~ ^-+(u|no-units) ]]; then
```

File: scripts/cromwell/get_cromwell_resource_usage2.sh [hunk 3]
```diff
@@ -5,17 +5,19 @@ function show_help() {
...
-      sent to stderr, so this is just syntactic sugar for file redirection.
+    WORKFLOW_INFO: specifies workflow identity. Can be
+                   a) a cromwell workflow ID
+                   b) a path to workflow output in google cloud (starting with gs://)
+                   c) a local path to workflow output
+    OPTIONS:
+        -r --raw-output
+            If set, output data as tab-separated table. Otherwise pass data through column -t for
+            easier reading.
+        -u --hide-units
```
Comment (epiercehoffman) [inline scripts/cromwell/get_cromwell_resource_usage2.sh:16]:
```suggestion
        -u --no-units
```

**Review by epiercehoffman — CHANGES_REQUESTED** [review]
I am so sorry it took me this long to review your changes again, especially since you responded to my last review so quickly! Thanks so much for adding the option to hide the second units header line and for making the first header line have unique column names - that will make parsing the output of this script in python simpler for sure. 

I tested a whole bunch of combinations of arguments and had no trouble until `--hide-units`, which gets matched by the `^-+(h|help)` regex before it gets to `^-+(u|hide-units)`. This is certainly not a big deal as `-u` works perfectly, but for the aesthetic I have suggested changing the longer-form option to hide the units to `--no-units`, as that seemed to be the easiest way to deal with this problem. If you are satisfied with that change, then I think this is all set to merge - I tested the `--no-units` change already locally so I don't feel the need to do another round of testing.

File: scripts/cromwell/get_cromwell_resource_usage2.sh (hunk 3 above)
Comment (TedBrookings) [reply scripts/cromwell/get_cromwell_resource_usage2.sh:16]: Good catch, and that's a good fix.

**Review by epiercehoffman — APPROVED** [review]

**Pull request merged**